File: activity14 Python/gdb/domain/fixed_deposit_account.py
# gdb/domain/fixed_deposit_account.py
from gdb.domain.abstract_account import AbstractAccount

class FixedDepositAccount(AbstractAccount):
    def __init__(self, account_number: str, name: str, age: int, balance: float, status: str = "Active", pin: str = "0000", tenure_months: int = 12, interest_rate: float = 6.5) -> None:
        super().__init__(account_number, name, age, balance, status, pin)
        self._tenure_months = tenure_months
        self._interest_rate = interest_rate

    def calculate_interest(self) -> float:
        return (self._balance * self._interest_rate * (self._tenure_months / 12.0)) / 100.0

    def get_account_type(self) -> str:
        return "FixedDeposit"

File: activity14 Java/src/com/gdb/domain/FixedDepositAccount.java
package com.gdb.domain;

import com.gdb.exceptions.*;

public class FixedDepositAccount extends AbstractAccount {
    private int tenureMonths;
    private double interestRate;

    public FixedDepositAccount(String accountNumber, String name, int age, double balance, String status, String pin, int tenureMonths, double interestRate) {
        super(accountNumber, name, age, balance, "FIXED_DEPOSIT", status, pin);
        this.tenureMonths = tenureMonths;
        this.interestRate = interestRate;
    }

    @Override
    public void processDebit(double amount) throws AccountException {
        throw new AccountException("Premature withdrawal not allowed on Fixed Deposit");
    }

    public double calculateMaturityAmount() {
        return this.balance * Math.pow(1 + (interestRate / 100.0) / 12, 12 * (tenureMonths / 12.0));
    }

    public int getTenureMonths() { return tenureMonths; }
    public double getInterestRate() { return interestRate; }
}
